homeassistant/config_entries.py

```diff
@@ -366,6 +377,22 @@ def __repr__(self) -> str:
             f"title={self.title} state={self.state} unique_id={self.unique_id}>"
         )
 
+    def __setattr__(self, key: str, value: Any) -> None:
+        """Set an attribute."""
```

Comment:
If entry id and unique id we need to fail hard since changing then can corrupt internal state 

the others can be a warning for now and fail hard after a deprecation period